DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py

```diff
@@ -0,0 +1,272 @@
+    tgen_utils_stop_protocols, tgen_utils_traffic_generator_connect,
+)
+
+
+pytestmark = pytest.mark.suite_functional_vlan
+
+packet_vids = ["X", 1, 2, 3, 4]  # VLAN tag number contained in the transmitted packet
+port_map = (
+    {"port": 0, "settings": [{"vlan": 2, "untagged": False, "pvid": False},
+                             {"vlan": 3, "untagged": True, "pvid": False},
+                             {"vlan": 4, "untagged": True, "pvid": True}]},
+    {"port": 1, "settings": [{"vlan": 2, "untagged": False, "pvid": False}]},
+    {"port": 2, "settings": [{"vlan": 3, "untagged": True, "pvid": True}]},
+    {"port": 3, "settings": [{"vlan": 4, "untagged": True, "pvid": True}]},
+)
+
+
+@pytest.mark.asyncio
+async def test_vlan_all_supported_modes_broadcast(testbed):
+    """
+    Test Name: Broadcast with all modes supported
+    Test Suite: suite_functional_vlan
+    Test Overview: Test VLAN with broadcast when all modes supported
+    Test Procedure:
+    1.Initiate test params.
+    2.Set links to vlans.
+    # 3.Map receiving and non receiving dut_ports.
+        Port 1 ->  tx_port
+        Ports 2, 3, 4 -> rx_ports
+    4.Setup packet stream(s) for the broadcast packet:
+    5.Send traffic to rx_ports
+         5.1 Verify traffic with vlan "x" received on port 4
+         5.2 verify traffic with vlan "1" dropped
+         5.3 Verify traffic with vlan "2" received on port 2
+         5.4 Verify traffic with vlan "3" received on port 3
+         5.5 Verify traffic with vlan "4" received on port 4
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2.Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3.Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[3]][0]["name"], dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {
+        f"traffic with VLAN ID: {vlan}": {
+            "type": "ethernetVlan",
```

Comment:
Are you sure that you are sending packets with broadcast mac? When using `ethernet`/`ethernetVlan` streams Ixia will choose the src and dst macs for you.